pkg/build/builder/docker.go

```diff
@@ -53,9 +56,15 @@ func (d *DockerBuilder) Build() error {
 	if err != nil {
 		return err
 	}
-	if err := fetchSource(buildDir, d.build, d.urlTimeout, os.Stdin, d.git); err != nil {
+	sourceInfo, err := fetchSource(buildDir, d.build, d.urlTimeout, os.Stdin, d.git)
+	if err != nil {
 		return err
 	}
+	if sourceInfo != nil {
+		if err := updateBuildRevision(d.client, d.build, sourceInfo); err != nil {
```

Comment:
The one drawback is that you can't create a real clone of that build, which I guess is no worse than where we are today.